plugins/mediatek-scaler/fu-mediatek-scaler-device.c

```diff
@@ -107,9 +113,39 @@ fu_mediatek_scaler_device_set_i2c_dev(FuMediatekScalerDevice *self,
 
 		/* the first i2c_dev is enforced to represent the dp aux device */
 		self->i2c_dev = g_object_ref(g_ptr_array_index(i2c_devs, 0));
-		g_debug("found I2C bus at %s, using this device",
-			fu_udev_device_get_sysfs_path(self->i2c_dev));
-		return fu_udev_device_set_physical_id(self->i2c_dev, "i2c", error);
+		syspath = fu_udev_device_get_sysfs_path(self->i2c_dev);
+		g_debug("found I2C bus at %s, using this device", syspath);
+
+		/* set device open flags */
+		fu_udev_device_add_open_flag(self->i2c_dev, FU_IO_CHANNEL_OPEN_FLAG_READ);
+		fu_udev_device_add_open_flag(self->i2c_dev, FU_IO_CHANNEL_OPEN_FLAG_WRITE);
+
+		/* get devpath by its sysfs */
+		udev_client = g_udev_client_new(NULL);
```

Comment:
I don't think it's a good idea to do this in plugin code -- the GUdevDevice that's returned from GUdevClient **A** can't be used from GUdevClient **B** -- and if you get it wrong everything explodes.

Reply:
in this case we take the devpath and set it to the `proxy` i2c-dev then drop it. How about let's do this PR to revive mt9700 first as it is broken now, and think about future improvements?